client/app/components/joinGroupList/joinGroupList.js

```diff
@@ -0,0 +1,36 @@
+import angular from "angular";
+import uiRouter from "@uirouter/angularjs";
+import Authentication from "../../services/authentication/authentication";
+import joinGroupListComponent from "./joinGroupList.component";
+
+let joinGroupListModule = angular.module("joinGroupList", [
+  uiRouter,
+  Authentication
+])
+
+.component("joinGroupList", joinGroupListComponent)
+
+.config(($stateProvider, hookProvider) => {
+  "ngInject";
+  $stateProvider
+    .state("joinGroupList", {
+      parent: "main",
+      url: "/joinGroupList",
```

Comment:
I'd like to avoid camel-case urls, I would prefer something like /join-group or /group/join